### PR TITLE
*: add table option: pre_split_regions

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
This is for pre-split region when create table with shard_row_id_bits.

### Check List 

Tests 

 - Unit test

Code changes

Side effects

### CODE REVIEW

File: misc.go [hunk 1]
```diff
@@ -408,6 +408,7 @@ var tokenMap = map[string]int{
 	"QUERIES":                  queries,
 	"QUICK":                    quick,
 	"SHARD_ROW_ID_BITS":        shardRowIDBits,
+	"PRE_SPLIT_REGIONS":        preSplitResions,
```
Comment (kennytm) [inline misc.go]:
```suggestion
	"PRE_SPLIT_REGIONS":        preSplitRegions,
```

Fix the PR title too.

File: parser.y [hunk 2]
```diff
@@ -203,6 +203,7 @@ import (
 	primary			"PRIMARY"
 	procedure		"PROCEDURE"
 	shardRowIDBits		"SHARD_ROW_ID_BITS"
+	preSplitResions		"PRE_SPLIT_REGIONS"
```
Comment (kennytm) [inline parser.y]:
```suggestion
	preSplitRegions		"PRE_SPLIT_REGIONS"
```

File: ast/ddl.go [hunk 3]
```diff
@@ -1326,6 +1327,10 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		ctx.WriteKeyWord("SHARD_ROW_ID_BITS ")
 		ctx.WritePlain("= ")
 		ctx.WritePlainf("%d", n.UintValue)
+	case TableOptionPreSplitRegion:
+		ctx.WriteKeyWord("PRE_SPLIT_REGIONS ")
+		ctx.WritePlain("= ")
+		ctx.WritePlainf("%d", n.UintValue)
```
Comment (kennytm) [inline ast/ddl.go]: Let's combine this with the statement above.

```go
		ctx.WritePlainf("= %d", n.UintValue)
```

Comment (crazycs520) [reply ast/ddl.go]: done.

**Pull request merged**